qiskit/transpiler/__init__.py

```diff
@@ -155,15 +228,599 @@
+circuit.
+
+The routing stage must set the properties ``final_layout`` and ``virtual_permutation_layout`` in
+the :class:`.PropertySet` if routing has taken place.
+
+All of Qiskit's built-in routing stages will additionally run the :class:`.VF2PostLayout` pass after
+routing.  This may choose to reassign the initial layout, if lower-error qubits can be found.  This
+pass is very similar to the :class:`.VF2Layout` class that :ref:`the default layout plugin
+<transpiler-preset-stage-layout-default>` uses, except in :class:`.VF2PostLayout` we can guarantee
+that there is at least one isomorphic induced subgraph of the target topology that matches the
+circuit topology.
+
+.. note::
+
+    Qiskit's built-in routing plugins all generally assume that all pairs of qubits with a
+    defined two-qubit link have a *universal* set of gates defined for those two qubits.  Hardware
+    does not necessarily need to respect this (for example, if the only defined two-qubit gate is
+    ``swap``, then entangling operations like ``cx`` cannot be realized), but Qiskit does not yet
+    consider this possibility.
+
+.. note::
+
+    Finding the minimal number of swaps to insert is known to be a non-polynomial problem.  This
+    means it is prohibitively expensive to attempt, so many of Qiskit's built-in algorithms are
+    stochastic, and you may see large variation between different compilations.  If you need
+    reproducibility, be sure to set the ``seed_transpiler`` argument of
+    :func:`.generate_preset_pass_manager` or :func:`.transpile`.
+
+When writing :ref:`stage plugins <transpiler-preset-stage-plugins>`, the entry point for ``routing``
+is ``qiskit.transpiler.routing``.  The built-in plugins are:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Method
+      - Summary
+
+    * - :ref:`sabre <transpiler-preset-stage-routing-sabre>`
+      - Default.  Uses `Qiskit's modified Sabre routing algorithm <sabre-lightsabre-paper_>`_ to swap
+        map.
+
+    * - :ref:`none <transpiler-preset-stage-routing-none>`
+      - Disable routing.  Raises an error if routing is required.
+
+    * - :ref:`basic <transpiler-preset-stage-routing-basic>`
+      - Greedy swap insertion to route a single operation at a time.
+
+    * - :ref:`stochastic <transpiler-preset-stage-routing-stochastic>`
+      - Consider operations layer-by-layer, using a stochastic algorithm to find swap networks that
+        implement a suitable permutation to make the layer executable.
+
+    * - :ref:`lookahead <transpiler-preset-stage-routing-lookahead>`
+      - Breadth-first search with heuristic pruning to find swaps that make gates executable.
+
+.. _transpiler-preset-stage-routing-none:
+
+Built-in ``none`` plugin
+........................
+
+A dummy plugin used to disable routing entirely.  This can occasionally be useful for
+hardware-configuration experiments, or in certain special cases of partial compilation.
```

Comment:
```suggestion
hardware configuration experiments, or in certain special cases of partial compilation.
```

Reply:
In this case, it is a "(hardware configuration) experiment", so I think the hyphen is right.